### ISSUE TITLE
RStudio crashes with igraph::estimate_betweenness

### ISSUE THREAD

**Issue by heinonmatti** [igraph/rigraph#415]
I was trying to create a reprex for another issue, but found out something that makes RStudio crash. 

This happened on my local machine, a clean RStudio on a virtual desktop, and a friend who run the same thing. Figured the lack of an error might imply C.

R 4.0.1, RStudio 1.3.959 

```
dummydata <- matrix(rnorm(n = 100),
                    ncol = 10,
                    nrow = 10)

diag(dummydata) <- NA

graph_object <- igraph::graph_from_adjacency_matrix(dummydata,
                                    mode = "upper",
                                    weighted = TRUE)

igraph::estimate_betweenness(graph_object, 
                             cutoff = 1/1e-10)
```

**Comment by szhorvat** [comment]
@vtraag This still segfaults on the `dev` branch

**Comment by vtraag** [comment]
The problem is actually unrelated to betweenness. If you check the weights you will see
```
 [1]          NA  0.01284737 -1.21653762 -0.23853134 -1.21382709 -0.34012105
 [7] -1.40364533 -0.88345347  0.57025979 -0.22345809          NA -0.76975153
[13] -0.50450507  0.55570354  0.80024502  0.19304418  0.95862552 -1.51834231
[19] -0.57626774          NA  1.25137490  0.58929874  1.79797192 -1.82578692
[25]  0.66203065 -1.35881106 -0.34228823          NA -0.13758046 -1.27231601
[31] -1.12024809 -0.84292438 -1.04521100 -0.37829784          NA -0.98374987
[37]  0.08184673  0.09688461 -0.71149631 -0.18395002          NA -0.51256498
[43] -0.71719939  0.19803773 -1.15646510          NA -0.06596688  1.37575669
[49] -0.69596753          NA -0.16668556 -0.94835176          NA -0.45375013
[55]          NA
```
These weights cannot be used in C, and should actually not be passed to the C core of `igraph`. This problem is addressed in PR https://github.com/igraph/rigraph/pull/393. This PR is yet to be merged, we should double check if that solution is correct.

**Comment by ntamas** [comment]
#393 seems okay to be in general, assuming that there are no other occurrences of the `any(!is.na(x))` construct in the source.

**Comment by szhorvat** [comment]
This is fixed in the 1.3 dev version, which can now be installed using `remotes::install_github("igraph/rigraph@master")` (assuming that the R development tools are installed).

It will now print an error:

```
Error in betweenness(graph, v = vids, directed = directed, cutoff = cutoff,  : 
  At core/centrality/betweenness.c:109 : Weight vector must not contain NaN values. Invalid value
```

@ntamas Can we close this now, or do you want to wait for the 1.3 release? It'd be good to close it now I think. Otherwise it's hard to keep track of what's been fixed.
